pkg/controllers/membercluster/v1beta1/membercluster_controller.go

```diff
@@ -121,86 +123,135 @@ func (r *Reconciler) handleDelete(ctx context.Context, mc *clusterv1beta1.Member
 		klog.V(2).InfoS("No need to do anything for the deleting member cluster without a finalizer", "memberCluster", mcObjRef)
 		return runtime.Result{}, nil
 	}
-	currentImc, err := r.getInternalMemberCluster(ctx, mc.GetName())
-	if err != nil {
-		return runtime.Result{}, err
+	// check if the namespace still exist
+	var currentNS corev1.Namespace
+	namespaceName := fmt.Sprintf(utils.NamespaceNameFormat, mc.Name)
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: namespaceName}, &currentNS); err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "Failed to get the member cluster namespace", "memberCluster", mcObjRef)
+			return runtime.Result{}, controller.NewAPIServerError(true, err)
+		}
+		klog.V(2).InfoS("The member cluster namespace is not found, remove the finalizer", "memberCluster", mcObjRef)
+		controllerutil.RemoveFinalizer(mc, placementv1beta1.MemberClusterFinalizer)
+		return runtime.Result{}, controller.NewUpdateIgnoreConflictError(r.Update(ctx, mc))
+	}
+	// check if the namespace is being deleted already, just wait for it to be deleted
+	if !currentNS.DeletionTimestamp.IsZero() {
+		klog.V(2).InfoS("The member cluster namespace is still being deleted", "memberCluster", mcObjRef, "deleteTimestamp", currentNS.DeletionTimestamp)
+		var stuckErr error
+		if time.Now().After(currentNS.DeletionTimestamp.Add(5 * time.Minute)) {
+			// alert if the namespace is stuck in deleting for more than 5 minutes
+			stuckErr = controller.NewUnexpectedBehaviorError(fmt.Errorf("the member cluster namespace %s has been deleting since %s", namespaceName, currentNS.DeletionTimestamp.Format(time.RFC3339)))
+		}
+		return runtime.Result{RequeueAfter: time.Second}, stuckErr
+	}
+	currentImc := &clusterv1beta1.InternalMemberCluster{}
+	imcNamespacedName := types.NamespacedName{Namespace: namespaceName, Name: mc.Name}
+	if err := r.Client.Get(ctx, imcNamespacedName, currentImc); err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.ErrorS(err, "Failed to get internal member cluster", "internalMemberCluster", imcNamespacedName)
+			return runtime.Result{}, controller.NewAPIServerError(true, err)
+		}
+		// this is possible since we garbage collect the internal member cluster first before deleting the MC
+		klog.V(2).Info("InternalMemberCluster not found, start garbage collecting", "memberCluster", mcObjRef)
+		return runtime.Result{Requeue: true}, r.garbageCollect(ctx, mc)
```

Comment:
we need to update the mc status here in case last time update mc call fails. Otherwise, the mc status is not reliable.

Reply:
We can't as the MC status is stale since there is no IMC.

Reply:
Should we treat the IMC not found as MC joined as false in this case?